themes/default/layouts/partials/head.html

```diff
@@ -111,7 +111,12 @@
         https://support.google.com/webmasters/answer/139066?hl=en
         https://en.wikipedia.org/wiki/Canonical_link_element
     -->
-    <link rel="canonical" href="{{ .Site.Params.canonicalURL }}{{ .RelPermalink }}" >
+
+    {{ if .Params.canonical_link }}
+        <link rel="canonical" href="{{ .Params.canonical_link }}" >
```

Comment:
Super minor, just for consistency:

```suggestion
    {{ if .Params.canonical_url }}
        <link rel="canonical" href="{{ .Params.canonical_url }}" >
```